egs/librispeech/asr/simple_v1/decode.py

```diff
@@ -0,0 +1,175 @@
+    return
+    # load L, G, symbol_table
+    lang_dir = 'data/lang_nosp'
+    symbol_table = k2.SymbolTable.from_file(lang_dir + '/words.txt')
+
+    if not os.path.exists(lang_dir + '/LG.pt'):
+        print("Loading L_disambig.fst.txt")
+        with open(lang_dir + '/L_disambig.fst.txt') as f:
+            L = k2.Fsa.from_openfst(f.read(), acceptor=False)
+        print("Loading G.fsa.txt")
+        with open(lang_dir + '/G.fsa.txt') as f:
+            G = k2.Fsa.from_openfst(f.read(), acceptor=True)
+        L = k2.arc_sort(L.invert_())
+        G = k2.arc_sort(G)
+        print("Intersecting L and G")
+        LG = k2.intersect(L, G)
+        print(LG.shape)
+        print("Connect L*G")
+        LG = k2.connect(LG.invert_())
+        print(LG.shape)
+        print("Determinize L*G")
+        LG = k2.determinize(LG)
+        print(LG.shape)
+        print("Connect L*G")
+        LG = k2.connect(LG)
+        print(LG.shape)
+        print("Remove disambiguation symbols on L*G")
+        LG.labels[LG.labels >= 347] = 0
+        LG.aux_labels[LG.aux_labels >= 200004] = 0
+        LG = k2.add_epsilon_self_loops(LG)
+        LG = k2.arc_sort(LG)
+        print((LG.properties & k2.fsa_properties.ARC_SORTED) != 0)
+        torch.save(LG.as_dict(), lang_dir + '/LG.pt')
+        # print(LG)
+    else:
+        d = torch.load(lang_dir + '/LG.pt')
+        print("Loading pre-prepared LG")
+        LG = k2.Fsa.from_dict(d)
+
+    return
+    # load dataset
+    feature_dir = 'exp/data'
+    print("About to get test cuts")
+    cuts_test = CutSet.from_json(feature_dir + '/cuts_test-clean.json.gz')
+
+    print("About to create test dataset")
+    test = K2SpeechRecognitionIterableDataset(cuts_test,
+                                              max_frames=100000,
+                                              shuffle=False)
+    print("About to create test dataloader")
+    test_dl = torch.utils.data.DataLoader(test, batch_size=None, num_workers=1)
+
+    exp_dir = 'exp'
+    setup_logger('{}/log/log-decode'.format(exp_dir))
+
+    if not torch.cuda.is_available():
+        logging.error('No GPU detected!')
+        sys.exit(-1)
+
+    print("About to load model")
+    device_id = 1
+    device = torch.device('cuda', device_id)
+    model = Model(num_features=40, num_classes=364)
+    checkpoint = os.path.join(exp_dir, 'epoch-9.pt')
+    load_checkpoint(checkpoint, model)
+    model.to(device)
+    model.eval()
+
+    LG = LG.to(device)
+    LG.scores.requires_grad_(False)
+    subsampling = 3  # must be kept in sync with model.
+    results = decode(dataloader=test_dl,
+                     model=model,
+                     subsampling=subsampling,
+                     device=device,
+                     LG=LG,
+                     symbols=symbol_table)
+    for ref, hyp in results:
+        print('ref=', ref, ', hyp=', hyp)
+    # compute WER
+    dist = sum(editdistance.eval(ref, hyp) for ref, hyp in results)
```

Comment:
Use “pip install kaldialign” for identical results with Kaldi

Reply:
Does it have a similar interface?  Would have to be imported as `import kaldialign`, right?

Reply:
Yeah. See usage here https://pypi.org/project/kaldialign/

Reply:
@danpovey I am not very much familiar with `editdistance`. I only vaguely remember evaluating several libs in the past and none of them got me the same WER/alignment as Kaldi, so I ported Kaldi's code to Python myself. A pleasant side effect of `kaldialign` is that the code is way faster (I compared to `Biopython` at that time, but `editdistance` says that it's a pure Python implementation so it's likely slow too). If you want we can just compare the resulting numbers for both libs.

Reply:
OK.  Using kaldialign is fine with me, sounds good.

Reply:
`editdistance` is implemented with C++, there's a benchmark here https://github.com/roy-ht/editdistance.
 Anyway, Just mention it. I'm OK with kaldialing as well.

Reply:
Oooh I looked up the wrong editdistance (there seem to be more than one libs with that name). Let's use whatever turns out to best.